distributed/tests/test_failed_workers.py

```diff
@@ -354,42 +349,41 @@ def __sizeof__(self) -> int:
 async def test_worker_who_has_clears_after_failed_connection(c, s, a, b):
     """This test is very sensitive to cluster state consistency. Timeouts often
     indicate subtle deadlocks. Be mindful when marking flaky/repeat/etc."""
-    n = await Nanny(s.address, nthreads=2, loop=s.loop)
+    async with Nanny(s.address, nthreads=2) as n:
+        while len(s.nthreads) < 3:
+            await asyncio.sleep(0.01)
 
-    while len(s.nthreads) < 3:
-        await asyncio.sleep(0.01)
+        def slow_ser(x, delay):
+            return SlowTransmitData(x, delay=delay)
 
-    def slow_ser(x, delay):
-        return SlowTransmitData(x, delay=delay)
-
-    n_worker_address = n.worker_address
-    futures = c.map(
-        slow_ser,
-        range(20),
-        delay=0.1,
-        key=["f%d" % i for i in range(20)],
-        workers=[n_worker_address],
-        allow_other_workers=True,
-    )
-
-    def sink(*args):
-        pass
+        n_worker_address = n.worker_address
+        futures = c.map(
+            slow_ser,
+            range(20),
+            delay=0.1,
+            key=["f%d" % i for i in range(20)],
+            workers=[n_worker_address],
+            allow_other_workers=True,
+        )
 
-    await wait(futures)
-    result_fut = c.submit(sink, futures, workers=a.address)
+        def sink(*args):
+            pass
 
-    with suppress(CommClosedError):
-        await c._run(os._exit, 1, workers=[n_worker_address])
+        await wait(futures)
+        result_fut = c.submit(sink, futures, workers=a.address)
 
-    while len(s.workers) > 2:
-        await asyncio.sleep(0.01)
+        with suppress(CommClosedError):
+            await c._run(os._exit, 1, workers=[n_worker_address])
 
-    await result_fut
+        while len(s.workers) > 2:
+            await asyncio.sleep(0.01)
 
-    assert not a.has_what.get(n_worker_address)
-    assert not any(n_worker_address in s for ts in a.tasks.values() for s in ts.who_has)
+        await result_fut
 
-    await n.close()
+        assert not a.has_what.get(n_worker_address)
+        assert not any(
+            n_worker_address in s for ts in a.tasks.values() for s in ts.who_has
+        )
```

Comment:
A local vscode diff indicates that these changes are merely indents (github doesn't do such a good job here).

Reply:
Yup.